src/main/java/de/tum/in/www1/artemis/web/rest/programming/ProgrammingExerciseResource.java

```diff
@@ -301,9 +301,14 @@ public ResponseEntity<ProgrammingExercise> updateProgrammingExercise(@RequestBod
         if (!Objects.equals(programmingExerciseBeforeUpdate.isTestwiseCoverageEnabled(), updatedProgrammingExercise.isTestwiseCoverageEnabled())) {
             throw new BadRequestAlertException("Testwise coverage enabled flag must not be changed", ENTITY_NAME, "testwiseCoverageCannotChange");
         }
```

Comment:
Should we also reject the update with an error in case `updatedProgrammingExercise.isAllowOnlineIde()` is `true` but the `theia` profile is not active on the server?

Reply:
Especially in the those cases a guard (as you mentioned in your other comment) would be super annoying. I think adding the guard would ~double the LOC throughout my changes